Patch-Source: https://gitlab.xfce.org/apps/xfce4-notifyd/-/merge_requests/101
---
From aff7cb87865c18ccc46d96fc57ccaa66f1771780 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Mon, 8 Dec 2025 14:22:27 +0100
Subject: [PATCH] build: allow systemd option without systemd installed

systemd.pc is just used to autodetect if the systemd option should be
enabled. If it is manually enabled with -Dsystemd=enabled, just ignore
the systemd dependency requirement and install the service.

Usefull in Alpine Linux where we package systemd files (e.g. for
postmarketOS) but do not ship systemd itself yet.
---
 meson.build               | 3 ++-
 xfce4-notifyd/meson.build | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 60351d1d..3549d209 100644
--- a/meson.build
+++ b/meson.build
@@ -47,7 +47,8 @@ gtk_layer_shell = dependency('gtk-layer-shell-0', version: dependency_versions['
 
 canberra_gtk3 = dependency('libcanberra-gtk3', version: dependency_versions['canberra'], required: get_option('sound'))
 
-systemd = dependency('systemd', version: dependency_versions['systemd'], required: get_option('systemd'))
+systemd = dependency('systemd', version: dependency_versions['systemd'], required: false)
+enable_systemd = (systemd.found() and get_option('systemd').allowed()) or get_option('systemd').enabled()
 
 feature_cflags = []
 
diff --git a/xfce4-notifyd/meson.build b/xfce4-notifyd/meson.build
index 948a2f9b..d2664475 100644
--- a/xfce4-notifyd/meson.build
+++ b/xfce4-notifyd/meson.build
@@ -66,7 +66,7 @@ executable(
 )
 
 systemd_service_dbus_text = ''
-if systemd.found()
+if enable_systemd
   systemd_service_dbus_text = 'SystemdService=xfce4-notifyd.service'
 endif
 daemon_service_data = configuration_data()
@@ -95,7 +95,7 @@ if get_option('dbus-start-daemon')
   )
 endif
 
-if systemd.found()
+if enable_systemd
   systemd_user_service_dir = get_option('systemd-user-service-dir')
   if systemd_user_service_dir == ''
     systemd_user_service_dir = get_option('prefix') / 'lib' / 'systemd' / 'user'
-- 
GitLab

